hostinfo/lsyncd.lua

```diff
@@ -0,0 +1,142 @@
+--[[
+Copyright 2015 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--]]
+local HostInfo = require('./base').HostInfo
+local Transform = require('stream').Transform
+local misc = require('./misc')
+local run = misc.run
+local safeMerge = misc.safeMerge
+local getInfoByVendor = misc.getInfoByVendor
+local async = require('async')
+--------------------------------------------------------------------------------------------------------------------
+local LsyncProcReader = Transform:extend()
+function LsyncProcReader:initialize()
+  Transform.initialize(self, {objectMode = true})
+end
+
+function LsyncProcReader:_transform(line, cb)
+  -- 'root     root     root     root     1 lsyncd /etc/lsyncd/lsyncd.conf.lua' -> '/etc/lsyncd/lsyncd.conf.lua'
+  local config = line:match('%slsyncd%s*(%S+)')
+  if config then self:push(config) end
+  cb()
+end
+--------------------------------------------------------------------------------------------------------------------
+--[[ Checks lsyncd ]]--
+local Info = HostInfo:extend()
+
+function Info:initialize()
+  HostInfo.initialize(self)
+end
+
+function Info:_run(callback)
+  local errTable, outTable, callbacked = {}, {}, false
+
+  local function finalCb()
+    if not callbacked then
+      callbacked = true
+      self:_pushParams(errTable, outTable)
+      callback()
+    end
+  end
+
+  local function getLsyncdBin(cb)
+    local lsyncPathDefault = '/usr/bin/lsyncd'
+    local options = {
+      ubuntu = '/usr/local/bin/lsyncd',
```

Comment:
perhaps use the default in this case to be: `/usr/bin/lsyncd', and then only set the ubuntu one.

Or, use fs.exist to check both paths and use the one that exists.